src/_pytest/doctest.py

```diff
@@ -401,6 +401,16 @@ class MockAwareDocTestFinder(doctest.DocTestFinder):
             https://bugs.python.org/issue25532
             """
 
+            def _find_lineno(self, obj, source_lines):
+                """
+                Doctest code does not take into account @property, hackish way to fix it.
+
+                https://bugs.python.org/issue17446
+                """
+                if inspect.isdatadescriptor(obj):
+                    obj = obj.fget
```

Comment:
this breaks for other descriptors
please check  for property explicitly or sort it differently

Reply:
@RonnyPfannschmidt thanks!
Would be good to have this covered in a test then.

Reply:
Thank you! I've also noticed it, working on a fix.

Reply:
Use the code from the PR maybe?  https://github.com/python/cpython/pull/3419

Reply:
Thank you for the suggestion! Indeed, I will replace the condition with an explicit check for the property. However, I did not find a way to write a doctest test for a case when the function/method is a data descriptor, but not property.